### PR TITLE
Restructure example layout

### PR DESCRIPTION
**Pull request by pepone**

### CODE REVIEW

**Review by externl — APPROVED** [review]
Looks good!

**Review by ReeceHumphreys — APPROVED** [review]

File: examples/ice/README.md [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+
+This folder contains example applications for using IceRPC + Protobuf integration.
+
+|                       |                                                                                      |
+|-----------------------|--------------------------------------------------------------------------------------|
+| [IceGrid](./IceGrid/) | Shows how an IceRPC client can call services hosted by servers managed by [IceGrid]. |
+| [Minimal](./Minimal/) | shows how an IceRPC client can call a service hosted by an [Ice] server.             |
+
+[IceGrid]: https://zeroc.com/products/ice
+[Minimal]: https://zeroc.com/products/ice/services/icegrid
```
Comment (InsertCreativityHere) [inline examples/ice/README.md:11]: I think these links got mixed up.
```suggestion
[Ice]: https://zeroc.com/products/ice
[IceGrid]: https://zeroc.com/products/ice/services/icegrid
```

File: examples/README.md [hunk 2]
```diff
@@ -27,28 +27,12 @@ of the NuGet packages, as described in [BUILDING].
...
-| [StreamProtobuf](./StreamProtobuf/)   | The Stream example updated to use Protobuf instead of Slice.                                                                        |
-| [Telemetry](./Telemetry/)             | Shows how to use the telemetry interceptor and middleware.                                                                          |
-| [Thermostat](./Thermostat/)           | Shows how to send requests via an intermediary server; includes sending requests the "other way around", from a server to a client. |
-| [Upload](./Upload/)                   | Shows how to upload a file from a client to a server by streaming this file.                                                        |
+|                         |                                                                            |
+|-------------------------|----------------------------------------------------------------------------|
+| [core](./core/)         | Contains examples that shows how to use the IceRPC core APIs.              |
```
Comment (InsertCreativityHere) [inline examples/README.md:32]:
```suggestion
| [core](./core/)         | Contains examples that show how to use the IceRPC core APIs.              |
```

File: examples/README.md [hunk 3]
```diff
@@ -27,28 +27,12 @@ of the NuGet packages, as described in [BUILDING].
...
+| [ice](./ice/)           | Contains examples that shows how IceRPC interoperates with [ZeroC Ice].    |
```
Comment (InsertCreativityHere) [inline examples/README.md:33]:
```suggestion
| [ice](./ice/)           | Contains examples that show how IceRPC interoperates with [ZeroC Ice].    |
```

File: examples/README.md [hunk 4]
```diff
@@ -27,28 +27,12 @@ of the NuGet packages, as described in [BUILDING].
...
+| [slice](./slice/)       | Contains examples that shows how to use the IceRPC + Slice integration.    |
```
Comment (InsertCreativityHere) [inline examples/README.md:34]:
```suggestion
| [slice](./slice/)       | Contains examples that show how to use the IceRPC + Slice integration.    |
```

File: examples/README.md [hunk 5]
```diff
@@ -27,28 +27,12 @@ of the NuGet packages, as described in [BUILDING].
...
+| [protobuf](./protobuf/) | Contains examples that shows how to use the IceRPC + Protobuf integration. |
```
Comment (InsertCreativityHere) [inline examples/README.md:35]:
```suggestion
| [protobuf](./protobuf/) | Contains examples that show how to use the IceRPC + Protobuf integration. |
```

**Review by InsertCreativityHere — APPROVED** [review]
I like it!

**Comment by pepone** [comment]
I will open an updated PR with feedback gathered in my blog

**Pull request closed without merging**